src/libraries/System.Private.CoreLib/src/System/ReadOnlySpan.cs

```diff
@@ -217,6 +217,11 @@ public bool IsEmpty
         /// </summary>
         public static ReadOnlySpan<T> Empty => default;
 
+        public static ReadOnlySpan<T> CastUp<TDerived>(ReadOnlySpan<TDerived> items) where TDerived : class?, T
```

Comment:
Please add XML doc comments describing the method. They'll be used to seed the docs.

Reply:
```suggestion
        /// <summary>
        /// Casts a read-only span of <typeparamref name="TDerived"/> to a read-only span of <typeparamref name="T"/>, allowing a covariant static cast to efficiently share the same memory.
        /// </summary>
        /// <typeparam name="TDerived">The element type of the source read-only span, which must be derived from <typeparamref name="T"/>.</typeparam>
        /// <param name="items">The source read-only span. No copy is made.</param>
        /// <returns>A read-only span with elements cast to the new type.</returns>
        public static ReadOnlySpan<T> CastUp<TDerived>(ReadOnlySpan<TDerived> items) where TDerived : class?, T
```
Thoughts @stephentoub?

Reply:
I like this better than the `ImmutableArray` version, which is
> Initializes a new instance of the [ImmutableArray<T>](https://learn.microsoft.com/en-us/dotnet/api/system.collections.immutable.immutablearray-1?view=net-8.0) struct based on the contents of an existing instance, allowing a covariant static cast to efficiently reuse the existing array.

and for `TDerived`
> The type of array element to return.

Reply:
I'd suggest moving the second phrase to be a sentence in remarks.  But otherwise, lgtm.

Reply:
```suggestion
        /// <summary>
        /// Casts a read-only span of <typeparamref name="TDerived"/> to a read-only span of <typeparamref name="T"/>.
        /// </summary>
        /// <typeparam name="TDerived">The element type of the source read-only span, which must be derived from <typeparamref name="T"/>.</typeparam>
        /// <param name="items">The source read-only span. No copy is made.</param>
        /// <returns>A read-only span with elements cast to the new type.</returns>
        /// <remarks>This method uses a covariant cast, producing a read-only span that shares the same memory as the source. The relationships expressed in the type constraints ensure that the cast is a safe operation.</remarks>
        public static ReadOnlySpan<T> CastUp<TDerived>(ReadOnlySpan<TDerived> items) where TDerived : class?, T
```